Terminal.Gui/Application/Application.cs

```diff
@@ -138,6 +138,8 @@ internal static List<CultureInfo> GetSupportedCultures ()
     // starts running and after Shutdown returns.
     internal static void ResetState (bool ignoreDisposed = false)
     {
+        Application.Navigation = new ApplicationNavigation ();
```

Comment:
Seems unsafe or at least unclean to just charge right into this without any conditions at all, no?

Reply:
I don't think so. ResetState is intended to be a very strong armed method. 

I keep going back and forth, though on the right semantics for objects like `ApplicationNavigation`, held by a property of a 'singleton static' like `Application`. Yes, I know your opinion on `Application` as a static, and I'd like to fix that eventually but not for v2. 

So... Given `Application` is a singleton-static and needs to be setup/reset for each new "app", what is the right pattern for ensuring non-static object refs it holds to be created/destroyed? 

Since `ApplicationNavigation` is actually used as a singleton, perhaps it should be implemented using the correct singleton pattern? Would appreciate your thoughts on this.

Reply:
Certainly the nuclear nature of it is understandable.

But what I specifically mean is that just reassigning a reference without ensuring cleanup in at least a naive way may not be ideal. I need to go back and look at the ApplicationNavigation class again, though, to see what (if anything) that is relevant for.

Reply:
As for the singleton pattern, I would ***strongly*** advise against continuing to make TG _more_ static than it already is.